### DISCUSSION TITLE
Problems with lc_geometry overflowing at periastron and not fitting ecc and per0

### DISCUSSION THREAD

**Discussion by rfreire-400** [phoebe-project/phoebe2#735] [Questions]
Hello,

So I've been working with the different estimators to see which gets me a better fit and I've been having some issues with lc geom. It says that primary is overflowing at periastron. I tried removing requivsumfrac from adopt_parameters in an attempt to resolve the issue but it did not work. Running a run_checks() tells me this:

<img width="795" alt="Screenshot 2023-05-17 at 11 51 19 AM" src="https://github.com/phoebe-project/phoebe2/assets/113457579/41e57091-a302-4172-94f6-18aae4cbb736">

Additionally, it does not want to fit ecc and per0 (I removed them from adopt_parameters for the image above). When I run_checks() with ecc and per0 in adopt_parameters, I get this:
<img width="795" alt="Screenshot 2023-05-17 at 11 53 22 AM" src="https://github.com/phoebe-project/phoebe2/assets/113457579/230cc849-b175-47c2-8f21-2a7ee52d8a53">

I tried flipping ecc and per0 with esinw and ecosw respectively and it still won't work:
<img width="795" alt="Screenshot 2023-05-17 at 11 55 22 AM" src="https://github.com/phoebe-project/phoebe2/assets/113457579/b32c2522-babd-42d8-b7bc-06704d26616f">

I'm currently running this using two-gaussian as the analytical_model. I tried doing it with polyfit, and had different issues. With polyfit, teffratio was a nan and had to be removed. I also had the following problems with run_checks():

<img width="795" alt="Screenshot 2023-05-17 at 12 02 14 PM" src="https://github.com/phoebe-project/phoebe2/assets/113457579/9e2eee9c-ea1d-485b-a2ae-1687f16a829c">

<img width="795" alt="Screenshot 2023-05-17 at 12 01 26 PM" src="https://github.com/phoebe-project/phoebe2/assets/113457579/2818f769-2ae6-4db0-9f0d-29330c2b78d2">

It's possible that lc geometry is struggling with this dataset as the system observed has tidally excited pulsations. The other estimators gave more promising values:
<img width="795" alt="Screenshot 2023-05-17 at 12 04 20 PM" src="https://github.com/phoebe-project/phoebe2/assets/113457579/dcf8ba20-0bde-4bc5-88d9-64295ae9272d">

I just wanted to make sure that I didn't do any mistakes using lc geometry before abandoning it and moving on with ebai. Any help would be greatly appreciated!

Thanks for your time,
Paco

**Comment by gecheline** [comment]
Hi Paco,
Lc_geometry is designed to work predominantly with clean binary light curves and depend on numerical "guesses" of where the eclipse edges are. The reason you're seeing it fail so badly in this case is because of the additional signal in the light curve that makes it difficult to identify the eclipses. Even the EBAI models I'm seeing, even though they work, are not truly separating the eclipses from the oscillations. 
This is overall a tricky dataset to model straigh-forwardly with phoebe, but there are ways you can go about it:
- before estimators and optimizers remove the additional signal and model only the eclipses. Find a suitable GP kernel to model the oscillations and add them back in when running samples to marginalize over them
- a lengthier process would be to manually fit the eclipses without removing the oscillations signal and go directly to optimizing and sampling the full dataset with an added GP kernel for the oscillations.

You can follow these tutorials on adding Gaussian Processes to the model: [docs example](http://phoebe-project.org/docs/2.4/examples/minimal_GPs), [workshop tutorial](http://phoebe-project.org/workshops/2022june/Tutorial_26_gaussian_processes).

**Reply by rfreire-400** [reply]
Hi gecheline,

Thank you for your help! I'm not sure I fully understand your comment. I was able to add a gaussian process to the model which gave a very good fit. I ended up using celerite2 since most of the noise came from the tidally excited pulsations.

<img width="712" alt="Screenshot 2023-05-19 at 10 04 58 AM" src="https://github.com/phoebe-project/phoebe2/assets/113457579/b3680991-4bf1-447f-9375-9b8c2da47cb7">

However, I am unsure where to go from here. Could you explain how to remove the additional signal and model only the eclipses? Do I have to manually figure out where the eclipses are or is there a method that can do it automatically? Sorry about the numerous questions, I'm still a bit new to this.

Thank you for your time,
Paco
